eole/utils/optimizers.py

```diff
@@ -41,17 +41,22 @@ def build_torch_optimizer(model, config):
     """
     params = [p for p in model.parameters() if p.requires_grad]
     betas = [config.adam_beta1, config.adam_beta2]
+    print(config.use_amp, optimi_available)
+    if config.use_amp or not optimi_available:
+        optim = torch_optim
+    else:
+        optim = torch_optimi
     if config.optim == "sgd":
         optimizer = optim.SGD(params, lr=config.learning_rate, weight_decay=config.weight_decay)
     elif config.optim == "adagrad":
-        optimizer = optim.Adagrad(
+        optimizer = torch.optim.Adagrad(
             params,
             lr=config.learning_rate,
             initial_accumulator_value=config.adagrad_accumulator_init,
             weight_decay=config.weight_decay,
         )
     elif config.optim == "adadelta":
-        optimizer = optim.Adadelta(params, lr=config.learning_rate, weight_decay=config.weight_decay)
+        optimizer = torch.optim.Adadelta(params, lr=config.learning_rate, weight_decay=config.weight_decay)
```

Comment:
Should we check against use_amp /optimi for such cases, to make it clear that such optimizers won't use the `optimi` based codepath?